sdk/program/src/stake/state.rs

```diff
@@ -16,13 +16,15 @@ use {
 
 pub type StakeActivationStatus = StakeHistoryEntry;
 
+pub type MustFullyActivateBeforeDeactivationIsPermittedFlag = u8;
+
 #[derive(Debug, Default, Serialize, Deserialize, PartialEq, Clone, Copy, AbiExample)]
 #[allow(clippy::large_enum_variant)]
 pub enum StakeState {
     #[default]
     Uninitialized,
     Initialized(Meta),
-    Stake(Meta, Stake),
+    Stake(Meta, Stake, u8),
```

Comment:
This is going to be a little yuck. I'm not feeling like we should modify `StakeState::Stake` though as that'll break backwards source compatibility (and I know I suggested that in the GHSA!) Instead I think we probably just pass the extra flag around explicitly.

Reply:
Should that flag be stored in the stake_account after redelegate? Then later on, deactivate load the account and use that flag to check for deactivation? I am not sure how we can pass that extra flag from relegate to deactivate?